jobsdb/jobsdb.go

```diff
@@ -2456,18 +2490,26 @@ func (jd *HandleT) refreshDSListLoop(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		}
-		start := time.Now()
-		jd.logger.Debugw("Start", "operation", "refreshDSListLoop")
-		timeoutCtx, cancel := context.WithTimeout(ctx, jd.refreshDSTimeout)
-		err := jd.dsListLock.WithLockInCtx(timeoutCtx, func(l lock.LockToken) error {
-			jd.refreshDSRangeList(l)
+		refresh := func() error {
+			var err error
+			start := time.Now()
+			defer stats.Default.NewTaggedStat("refresh_ds_loop", stats.TimerType, stats.Tags{"customVal": jd.tablePrefix, "error": strconv.FormatBool(err != nil)}).Since(start)
```

Comment:
Wrap it in a function, now it will use the `nil` error always
https://go.dev/play/p/FU4vdYWJeTF